content/blog/cumundi-guest-post/index.md

```diff
@@ -0,0 +1,314 @@
+---
+title: "Refactoring Pulumi Code with `aliases`"
+date: 2020-03-26
+meta_desc: "Cumundi uses Pulumi to build infrastructure components for their customers"
+meta_image: cumundi-pulumi.png
+authors:
+    - ringo-de-smet
+tags:
+    - Test Driven Developmnet
+---
+
+**Guest Article:** [Ringo De Smet](https://www.linkedin.com/in/ringodesmet/), Founder of [Cumundi](https://www.cumundi.cloud), standardizes on Pulumi for writing infrastructure as code libraries to deliver reusable building blocks. Pulumi enables him to iterate through the build-test-release cycle of these building blocks rapidly.
+
+Cumundi helps companies adopt the cloud more integrally. Companies are rebuilding the same 75% of the underlying infrastructure. But the shortage of people and time has led to limited investment in integrating all the non-functional requirements needed to run applications in the cloud.
+
+<!--more-->
+
+## How Pulumi Code Keeps Up with Changing Requirements
+
+At Cumundi, we build reusable libraries for our customers to set up their infrastructure integrated with best practices. These best practices span the range of non-functional requirements, which are most of the time not taken up with the same focus as application feature development. Where these non-functional requirements can be integrated into infrastructure setups, we include these in our code libraries. Cumundi focuses on all three major cloud providers to provide application-focused infrastructure libraries.
+
+For our internal infrastructure, it is not surprising that we also use Pulumi. At this early phase of our company (founded January 1st!), it is hard to foresee how we will evolve. With Pulumi, we write code in a modern programming language. A good coding practice is not to [optimize prematurely](http://wiki.c2.com/?PrematureOptimization) because the future is unpredictable. Another coding best practice is [red-green-refactor](https://en.wikipedia.org/wiki/Test-driven_development) or formally known as Test Driven Development. You take the current code, write a test for the new requirement, which initially fails (red). Next, you implement the code in the most straightforward way to make the test succeed (green) and you complete the cycle by refactoring the code to keep the design in a proper shape.
```

Comment:
```suggestion
It may come as no surprise that we also use Pulumi to manage Cumundi's own infrastructure. As a young company (founded January 1st!), it is hard to foresee how our infrastructure needs will evolve. With Pulumi, we write code in a modern programming language, and, good coding practice is not to [optimize prematurely](http://wiki.c2.com/?PrematureOptimization) because the future is unpredictable. Another coding best practice is [red-green-refactor](https://en.wikipedia.org/wiki/Test-driven_development) or formally known as Test-Driven Development. You take the current code, write a test for the new requirement, which initially fails (red). Next, you implement the code in the most straightforward way to make the test succeed (green) and you complete the cycle by refactoring the code to keep the design in proper shape.
```